modules/imgproc/src/filter.simd.hpp

```diff
@@ -465,6 +465,52 @@ struct RowVec_8u32s
     bool smallValues;
 };
 
+struct RowVec_8u32f
+{
+    RowVec_8u32f() {}
+    RowVec_8u32f( const Mat& _kernel ) { kernel = _kernel; }
+
+    int operator()(const uchar* _src, uchar* _dst, int width, int cn) const
+    {
+        CV_INSTRUMENT_REGION();
+
+        int i = 0, k, _ksize = kernel.rows + kernel.cols - 1;
+        float* dst = (float*)_dst;
+        const float* _kx = kernel.ptr<float>();
+        width *= cn;
+        for( ; i < width - v_uint8::nlanes; i += v_uint8::nlanes )
+        {
+            v_float32 s0 = vx_setzero_f32();
+            v_float32 s1 = vx_setzero_f32();
+            v_float32 s2 = vx_setzero_f32();
+            v_float32 s3 = vx_setzero_f32();
+            k = 0;
+            for( ; k < _ksize ; k++ )
+            {
+                v_float32 f = vx_setall_f32(_kx[k]);
+                const uchar* src = (const uchar*)_src + i + k * cn;
+                v_uint8 vs = vx_load(src);
+                v_int16 vs_l = v_reinterpret_as_s16(v_expand_low(vs));
+                v_int16 vs_h = v_reinterpret_as_s16(v_expand_high(vs));
+                v_float32 fvs_ll = v_cvt_f32(v_expand_low(vs_l));
+                v_float32 fvs_lh = v_cvt_f32(v_expand_high(vs_l));
+                v_float32 fvs_hl = v_cvt_f32(v_expand_low(vs_h));
+                v_float32 fvs_hh = v_cvt_f32(v_expand_high(vs_h));
+                s0 = v_muladd(fvs_ll, f, s0);
+                s1 = v_muladd(fvs_lh, f, s1);
+                s2 = v_muladd(fvs_hl, f, s2);
+                s3 = v_muladd(fvs_hh, f, s3);
+            }
+            v_store(dst + i, s0);
+            v_store(dst + i + v_float32::nlanes, s1);
+            v_store(dst + i + 2*v_float32::nlanes, s2);
+            v_store(dst + i + 3*v_float32::nlanes, s3);
+        }
+        return i;
+    }
+
+    Mat kernel;
```

Comment:
const reference + initializer list in ctor ?

Reply:
Fixed